src/elements/components/TextNodes.tsx

```diff
@@ -82,56 +84,87 @@ function TextNodes({
       });
     } else if (actions.length > 0) delta = applyNewDelta(delta);
 
+    // If text_mode property is set to 'data', then we don't want to render the text_formatted
+    // property, instead we the text from the data element specified in the text_source property
+    let textFromData = null;
+    let textSource = element.properties.text_source ?? '';
+    // convert to valid jsonpath
+    if (textSource.startsWith('feathery'))
+      textSource = textSource.replace('feathery', '$');
+
+    if (element.properties.text_mode === 'data') {
+      try {
+        textFromData = jsonpath.value(featheryContext, textSource);
+      } catch (e) {
+        // Ignoring errors due to things like unset text_source, deleted products etc.
+      }
+    }
+
     return (
       <span
         id={`span-${element.id}`}
         ref={spanRef}
         {...editableProps}
         key={text}
       >
-        {delta
-          .filter((op) => !!op.insert)
-          .map((op, i) => {
-            const attrs = op.attributes || {};
-            let onClick = () => {};
-            let cursor = 'inherit';
-            if (!editMode && !disabled) {
-              if (attrs.font_link) {
-                const link = replaceTextVariables(
-                  attrs.font_link,
-                  element.repeat
-                );
-                onClick = () => openTab(link);
-                cursor = 'pointer';
-              } else if (
-                attrs.fullSpan ||
-                (isNum(attrs.start) && isNum(attrs.end))
-              ) {
-                onClick = () => textSpanOnClick(attrs.start, attrs.end);
-                cursor = 'pointer';
+        {element.properties.text_mode === 'data' && textFromData !== null ? (
+          <span
```

Comment:
this span is basically the same as the span on line 152, except it's missing the onClick & cursor style (which i think is a bug). We can probably define these spans as the same component